rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/api/RecoveryCoordinator.java

```diff
@@ -152,4 +153,48 @@ public String replaceCompensator(
     public List<LRAStatusHolder> getRecoveringLRAs() {
         return lraService.getAllRecovering(true);
     }
+
+    @GET
+    @Path("failed")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "List failed Long Running Actions",
+            description = "Returns LRAs that have failed. " +
+                    " Failure records are vital pieces of data needed to aid failure tracking and analysis " +
+                    " and are retained for inspection.")
+    @APIResponse(responseCode = "200",
+            content = @Content(schema = @Schema(type = SchemaType.ARRAY, implementation = LRAStatusHolder.class)))
+    public List<LRAStatusHolder> getFailedLRAs() {
+        return lraService.getFailedLRAs();
+    }
+
+    @DELETE
+    @Path("{LraId}")
+    @Operation(summary = "Remove the log for an LRA")
+    @APIResponses({
+            @APIResponse(responseCode = "204",
+                    description = "If the LRA was successfully removed"),
+            @APIResponse(responseCode = "404",
+                    description = "The LRA was not recognised"),
+            @APIResponse(responseCode = "500",
+                    description = "If the input LRA does not correspond to a valid URI " +
+                            "(in which case the response entity will contain the error message) " +
+                            "or if the attempt to remove the LRA failed")
+    })
```

Comment:
So what are you asking for?
Are you saying that we should throw an error if the record is not in the failed state. If so I will reword the API and update the implementation.

Reply:
412 precondition failed would be more appropriate I think

Reply:
So are you saying that we should return 412 if the record is not in a failed state. If so I will reword the API and update the implementation (since in progress LRAs should not be deleted)?

Reply:
@mmusgrov yes, exactly. Thanks.

Reply:
Thanks, now I know how to code it up.

Reply:
@xstefank I added OpenAPI APIResponse annotations for 204, 412, 412 and 500 (and I had to make some updates to LRARecoveryModule so this code can distinguish between them).